lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene87/Lucene87Codec.java

```diff
@@ -101,14 +102,22 @@ public DocValuesFormat getDocValuesFormatForField(String field) {
 
   /** Instantiates a new codec. */
   public Lucene87Codec() {
+    this(Mode.BEST_COMPRESSION);
```

Comment:
Maybe we could avoid adding a default constructor here? I think it's only used in `Lucene87RWCodec`.

Reply:
We actually need the default constructor in codecs or you get errors like:

```
org.apache.lucene.codecs.Codec: org.apache.lucene.backward_codecs.lucene87.Lucene87Codec Unable to get public no-arg constructor
java.util.ServiceConfigurationError: org.apache.lucene.codecs.Codec: org.apache.lucene.backward_codecs.lucene87.Lucene87Codec Unable to get public no-arg constructor
```